### DISCUSSION TITLE
Can interactive stories be run in a CI pipeline?

### DISCUSSION THREAD

**Discussion by amuttsch** [storybookjs/storybook#16612] [Help] [answered]
Hi,

I just read the blog post about the interactive stories (https://storybook.js.org/blog/interactive-stories-beta/). This is a very cool feature and I'm looking forward to try it. Reading the blog post left me with a couple of questions I'd like to ask here:

- Can the interactive stories replace my jest unit tests for components? I imagine it would be cool to write them inside storybook and let it run in a real browser. 
- Can I write multiple play functions for a single story?
- Can I run the play functions in a CI with a headless browser similar to cypress? I image a command like `run-storybook-play` that fails if a play function fails?

Thanks!

**Comment by shilman** [comment, marked as answer]
> Can the interactive stories replace my jest unit tests for components? I imagine it would be cool to write them inside storybook and let it run in a real browser.

Coming in 6.5

> Can I write multiple play functions for a single story?

No. It's part of the motivation behind [CSF3](https://storybook.js.org/blog/component-story-format-3-0/). Rather than assigning multiple play functions for a story, just make a new story with a new play function and the rest of the settings identical:

```js
export const SomeStory = {
  args: { ... },
  parameters: { .... },
  play: async () => { ... },
}

export const AnotherStory = {
  ...SomeStory,
  play: async () => { ... }
}
```

> Can I run the play functions in a CI with a headless browser similar to cypress? I image a command like run-storybook-play that fails if a play function fails?

Coming in 6.5

**Reply by amuttsch** [reply]
Thanks for your reply, this answers my questions. Looking forward to it :-)
